app/controllers/concerns/sufia/files_controller_behavior.rb

```diff
@@ -228,6 +233,7 @@ def process_file(file)
         update_metadata_from_upload_screen
         actor.create_metadata(params[:batch_id])
         if actor.create_content(file, file.original_filename, file_path, file.content_type)
+          add_file_to_collection
```

Comment:
I think the actor should perform this step as part of the `#create_content` method.